### DISCUSSION TITLE
yform callback action - der Funktion parameter mitgeben

### DISCUSSION THREAD

**Discussion by tyrant88** [FriendsOfREDAXO/.github#5] [Q&A] [answered]
Wie kann man bei einer yform callback action der callback-Funktion Parameter übergeben?

`$yform->setActionField('callback', array('sh_gruppe_aenderungswunsch::newChange'));`

**Comment by gharlan** [comment, marked as answer]
Ich würde dann eine anonyme Funktion drum rum setzen, und darin den eigentlichen Aufruf machen. Also so:

```php
// php >= 7.4
$yform->setActionField('callback', [fn () => sh_gruppe_aenderungswunsch::newChange('hier', 'die', 'params')]);

// für php < 7.4 ohne die arrow function
$yform->setActionField('callback', [function () {
    sh_gruppe_aenderungswunsch::newChange('hier', 'die', 'params');
}]);
```

Hinweis: Default übergibt YForm bereits einen Parameter, das `rex_yform_action_callback`-Objekt. Das wird bei der obigen Variante jetzt verschluckt. Falls das Objekt weiterhin auch übergeben werden soll, dann so in der Art:

```php
$yform->setActionField('callback', [fn (rex_yform_action_callback $action) => sh_gruppe_aenderungswunsch::newChange($action, 'hier', 'die', 'params')]);
```

**Reply by tyrant88** [reply]
An anoyme Funktion kann ich mich sogar erinnern, stimmt. Ich dachte nur man muss einen String übergeben, aber dass da auch eine anonyme Funktion geht wusste ich nicht mehr. (Jetzt gesehen: "callable")
Wie elegant.

**Reply by tyrant88** [reply]
Danke dir!

**Comment by tyrant88** [comment]
Rein interessehalber: 
Wie würde man das per default übergebene $this Objekt ansprechen in der callback-Funktion?

**Reply by tyrant88** [reply]
func_get_args() ?

**Reply by gharlan** [reply]
Also bei deiner Variante im Ursprungsbeitrag und bei meiner dritten Variante würde es ja einfach als erster Param in deiner `newChange`-Methode ankommen:

```php
class sh_gruppe_aenderungswunsch
{
    static function newChange(rex_yform_action_callback $action)
    {
        // $action ist das "$this"-Objekt
    }
}
```

Bei den anderen Varianten von mir wird es verschluckt und du kommst nicht dran.

**Reply by tyrant88** [reply]
Hast mich erwischt, ich hätte einfach nur genauer hingucken müssen!
